src/message/MessageContainer.js

```diff
@@ -64,7 +64,7 @@ export default class MessageContainer extends React.PureComponent {
       pushRoute,
     } = this.props;
     const MessageComponent = isBrief ? MessageBrief : MessageFull;
-    const childrenNodes = htmlToDomTree(message.match_content || message.content);
+    const childrenNodes = htmlToDomTree((message.match_content || message.content).replace(/\r\n/g, '<br />').replace(/[\r\n]/g, '<br />'));
```

Comment:
I think it's actually a better design, if we're going to do these sorts of messy translations, to do them in the mobile client, because then it can be versioned along with the rest of the app's rendering logic.  

If we eventually decide on another interface than HTML that we want to use for sending the tree to the client with a clear specification, I could imagine doing that on the server side, but it'd really only be constructive if we think we'll be able to add new features to the markup with only changes on the server side.  If we have to do work on the mobile side anyway, it's better to have full control there, since then we need to spend less effort on compatibility with different server versions.